### PR TITLE
Feature/nex 519/add transactional

### PR DESCRIPTION
**Pull request by julien-sebire**
TaskQueue brokers cleanup:
- removed useless transaction handling for single persistence query
- added exception message
- removed dead code

This is required by https://github.com/oat-sa/extension-tao-task-queue/pull/104

### CODE REVIEW

**Review by mike-ionut-mihai-sandu-tao — CHANGES_REQUESTED** [review]
please check the failing tests

File: models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php [hunk 1]
```diff
@@ -20,11 +20,13 @@
 
 namespace oat\tao\model\taskQueue\TaskLog\Broker;
 
+use common_report_Report as Report;
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\ParameterType;
-use oat\oatbox\PhpSerializable;
-use common_report_Report as Report;
 use Doctrine\DBAL\Query\QueryBuilder;
+use Google\Cloud\Spanner\Transaction;
```
[the remark this answers is not in the document]
Comment (julien-sebire) [reply models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php]: As stated yesterday evening, we need a level of abstraction to be able to merge back to develop, but for the Spanner braznch, we can afford to keep it this way. The abstraction would just use a Google\Cloud\Spanner\Transaction for Spanner and a Doctrine\Dbal\Connection for other Rds, proxying the few methods we need (execute and commit).

File: models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php [hunk 2]
```diff
@@ -367,24 +368,17 @@ public function cancelCollection(CollectionInterface $collection)
      */
     public function deleteById($taskId)
     {
-        $this->getPersistence()->getPlatform()->beginTransaction();
+        $qb = $this->getQueryBuilder()
```
[the remark this answers is not in the document]
Comment (julien-sebire) [reply models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php]: For the sake of clarity on what can actually throw an exception.

File: models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php [hunk 3]
```diff
@@ -429,28 +423,27 @@ private function buildCounterStatusSql($statusColumn, array $inStatuses)
      */
     private function updateCollectionStatus(CollectionInterface $collection, $status)
     {
-        $this->getPersistence()->getPlatform()->beginTransaction();
-
+        $qb = $this->getQueryBuilder()
```
[the remark this answers is not in the document]
Comment (julien-sebire) [reply models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php]: Why would we? The idea is to limit the number of statements that can produce an exception. Exception have to be treated as a method to deal with things not going the supposed way, not a way to deal with a second king of result type.

**Review by julien-sebire — COMMENTED** [review]
ANswered your questions...

**Pull request closed without merging**